### ISSUE TITLE
a11y audit: Tile a11y fixes

### ISSUE THREAD

**Issue by marijohannessen** [IBM/carbon-components-react#624]
### Detailed description

- [ ] **Selectable tile**: The id tile-1 referenced by the 'for' attribute of a label element is not a valid form input element or an element with id tile-1 is either null or does not exist.
![screen shot 2018-02-15 at 4 10 16 pm](https://user-images.githubusercontent.com/5447411/36283680-ceeac588-126a-11e8-9c45-d8d46c13d8a7.png)

- [ ] Needs to focus on the icon in order to expand the tile content

- [ ] Unable to tab to Selectable & Clickable Tile
